service/history/timer_queue_standby_task_executor.go

```diff
@@ -472,6 +472,64 @@ func (t *timerQueueStandbyTaskExecutor) executeWorkflowExecutionTimeoutTask(
 		),
 	)
 }
+func (t *timerQueueStandbyTaskExecutor) executeStateMachineTimerTask(
+	ctx context.Context,
+	timerTask *tasks.StateMachineTimerTask,
+) error {
+	actionFn := func(
+		ctx context.Context,
+		wfContext workflow.Context,
+		mutableState workflow.MutableState,
+	) (any, error) {
+		processedTimers, err := t.executeStateMachineTimers(
+			ctx,
+			mutableState,
+			func(node *hsm.Node, task hsm.Task) error {
+				if task.Concurrent() {
+					//nolint:revive // concurrent tasks implements hsm.ConcurrentTask interface
+					concurrentTask := task.(hsm.ConcurrentTask)
+					return concurrentTask.Validate(node)
+				}
+				// If the task is expired and still valid in the standby queue,
+				// then the state machine is stale.
+				return consts.ErrTaskRetry
+			},
+		)
+		if err != nil {
+			if errors.Is(err, consts.ErrTaskRetry) {
+				// This handles the ErrTaskRetry error returned by executeStateMachineTimers.
+				return getHistoryResendInfo(mutableState)
+			}
+			return nil, err
+		}
+
+		// We haven't done any work, return without committing.
+		if processedTimers == 0 {
+			return nil, nil
+		}
+
+		if mutableState.GetExecutionState().State == enumsspb.WORKFLOW_EXECUTION_STATE_COMPLETED {
+			// Can't use UpdateWorkflowExecutionAsActive since it updates the current run, and we are operating on a
+			// closed workflow.
```

Comment:
Not sure this comment is relevant, but at minimum you'll want to `s/Active/Passive/`.